homeassistant/components/intent/timers.py

```diff
@@ -0,0 +1,772 @@
+        if timer is None:
+            return
+
+        # Capture event data before cancellation to get accurate seconds_left
+        event_data = timer.to_event()
+
+        timer.seconds = 0
+        timer.updated_at = time.monotonic_ns()
+        if not timer.is_paused:
+            timer.task.cancel()
+
+        self.hass.bus.async_fire(EVENT_INTENT_TIMER_CANCELLED, event_data)
+
+        _LOGGER.debug(
+            "Timer cancelled: id=%s, name=%s, seconds_left=%s, device_id=%s",
+            timer_id,
+            timer.name,
+            timer.seconds_left,
+            timer.device_id,
+        )
+
+    async def add_time(self, timer_id: str, seconds: int) -> None:
+        """Add time to a timer."""
+        if seconds == 0:
+            return
+
+        timer = self.timers.get(timer_id)
+        if timer is None:
+            return
+
+        timer.seconds = max(0, timer.seconds_left + seconds)
+        timer.updated_at = time.monotonic_ns()
+        if not timer.is_paused:
+            timer.task.cancel()
+            timer.task = self.hass.async_create_background_task(
+                self._wait_for_timer(timer_id, timer.seconds, timer.updated_at),
+                name=f"Timer {timer_id}",
+            )
+
+        self.hass.bus.async_fire(EVENT_INTENT_TIMER_UPDATED, timer.to_event())
+
+        if seconds > 0:
+            log_verb = "increased"
+            log_seconds = seconds
+        else:
+            log_verb = "decrease"
+            log_seconds = -seconds
+
+        _LOGGER.debug(
+            "Timer %s by %s second(s): id=%s, name=%s, seconds_left=%s, device_id=%s",
+            log_verb,
+            log_seconds,
+            timer_id,
+            timer.name,
+            timer.seconds_left,
+            timer.device_id,
+        )
+
+    async def remove_time(self, timer_id: str, seconds: int) -> None:
+        """Remove time from a timer."""
+        await self.add_time(timer_id, -seconds)
+
+    async def pause_timer(self, timer_id: str) -> None:
+        """Pauses a timer."""
+        timer = self.timers.get(timer_id)
+        if (timer is None) or timer.is_paused:
+            return
+
+        timer.seconds = timer.seconds_left
+        timer.updated_at = time.monotonic_ns()
+        timer.is_paused = True
+        timer.task.cancel()
+
+        self.hass.bus.async_fire(EVENT_INTENT_TIMER_UPDATED, timer.to_event())
+
+        _LOGGER.debug(
+            "Timer paused: id=%s, name=%s, seconds_left=%s, device_id=%s",
+            timer_id,
+            timer.name,
+            timer.seconds_left,
+            timer.device_id,
+        )
+
+    async def unpause_timer(self, timer_id: str) -> None:
+        """Unpause a timer."""
+        timer = self.timers.get(timer_id)
+        if (timer is None) or (not timer.is_paused):
+            return
+
+        timer.is_paused = False
+        timer.updated_at = time.monotonic_ns()
+        timer.task = self.hass.async_create_background_task(
```

Comment:
this should be a helper inside the timer class. Probably all moved to timer method.

Reply:
I can't move the task creation because it uses the `_timer_finished` method on the timer manager, which calls the event handlers. The timer object would need to manipulate the manager, which seems awkward.

Reply:
Moved the tasks off the timer object and into the manager.